### PR TITLE
Improve Studio Token Authentication URI Handler

### PR DESCRIPTION
**Pull request by julieg18**
Details are inline. 

Related to #3864

### CODE REVIEW

File: extension/src/vscode/external.ts [hunk 1]
```diff
@@ -10,12 +10,11 @@ export const getCallBackUrl = async (path: string) => {
   return uri.toString()
 }
 
-export const waitForUriResponse = (path: string, onResponse: () => unknown) => {
+export const waitForUriResponse = (path: string, onResponse: () => unknown) =>
   window.registerUriHandler({
     handleUri(uri: Uri): ProviderResult<void> {
-      if (uri.path.startsWith(path)) {
+      if (uri.path === path) {
```
Comment (julieg18) [inline extension/src/vscode/external.ts:16]: Studio is now decoding the uri so we can use the equals check.

File: extension/src/setup/studio.ts [hunk 2]
```diff
@@ -134,9 +137,12 @@ export class Studio extends Disposable {
     verificationUrlWithParams.searchParams.append('code', userCode)
 
     await openUrl(verificationUrlWithParams.toString())
-    void waitForUriResponse('/studio-complete-auth', () => {
-      void this.requestStudioToken(deviceCode, tokenUri)
-    })
+    this.studioAccessTokenUriHandler = waitForUriResponse(
```
Comment (julieg18) [inline extension/src/setup/studio.ts:140]: Realized we aren't disposing the uri handler after use. I updated the code to dispose of the listener if:

* we get a response from Studio
* the user wants to try authentication from the start again

File: extension/src/setup/studio.ts [hunk 3]
```diff
@@ -108,6 +109,8 @@ export class Studio extends Disposable {
   }
 
   public async requestStudioTokenAuthentication() {
+    this.studioAccessTokenUriHandler?.dispose()
```
Comment (mattseddon) [inline extension/src/setup/studio.ts:112]: [C] personal preference but I always set class members to `undefined` after disposing.

Comment (mattseddon) [reply extension/src/setup/studio.ts:112]: could create a `resetAccessTokenUriHandler` method

**Review by sroy3 — APPROVED** [review]

File: extension/src/setup/studio.ts (hunk 3 above)
Comment (julieg18) [reply extension/src/setup/studio.ts]: Done!

**Pull request merged**